packages/util/src/errors.ts

```diff
@@ -54,115 +54,93 @@
  *     }
  *   }
  */
-export type ErrorList<T> = { [code: string]: string };
+export type ErrorList<T extends string = string> = {
```

Comment:
Nice. Interesting that string enum satisfies the type constraint.

Reply:
Can you please apply this type across the repo?

Reply:
What do you mean by apply this type? This should be compatible with the previous version, and the compiler isn't complaining, so I don't think we need to change anything anywhere else?

Reply:
For example, https://github.com/firebase/firebase-js-sdk/blob/master/packages/app/src/errors.ts#L29, we can use this new type instead.

Reply:
Ah I see, sure, done :slightly_smiling_face:

Looks like `ErrorFactory` is only used in App and Messaging :thinking:

Reply:
I removed the default `string` from `ErrorList`, so if you haven't typed your errors with an enum or a union type, you'd have to use `ErrorList` like `ErrorList<string>` explicitly, instead of just `ErrorList`. This doesn't break anything as `ErrorList` wasn't being used anywhere.